src/TokensTracker/Trackers/TokenTransfer.cs

```diff
@@ -0,0 +1,36 @@
+using Neo.IO;
+using System.IO;
+using System.Numerics;
+
+namespace Neo.Plugins.Trackers
+{
+    public class TokenTransfer : ISerializable
+    {
+        public UInt160 UserScriptHash;
+        public uint BlockIndex;
+        public UInt256 TxHash;
+        public BigInteger Amount;
+
+        int ISerializable.Size =>
+            UInt160.Length +        // UserScriptHash
+            sizeof(uint) +          // BlockIndex
+            UInt256.Length +        // TxHash
+            Amount.GetVarSize();  // Amount
+
+        void ISerializable.Serialize(BinaryWriter writer)
+        {
+            writer.Write(UserScriptHash);
+            writer.Write(BlockIndex);
+            writer.Write(TxHash);
+            writer.WriteVarBytes(Amount.ToByteArray());
+        }
+
+        void ISerializable.Deserialize(BinaryReader reader)
+        {
+            UserScriptHash = reader.ReadSerializable<UInt160>();
+            BlockIndex = reader.ReadUInt32();
+            TxHash = reader.ReadSerializable<UInt256>();
+            Amount = new BigInteger(reader.ReadVarBytes(32));
```

Comment:
If we check the format in `Deserialize`, and not in `Serialize` we can deny the service during the reader, if someone want to deny the service should be skipped, and not stored. (same for TokenBalance, and the others)

Reply:
Could you show a more detail example?

Reply:
We should prevent deseralization problems because we can't trust in the source, so we should ensure that the input is valid for be deserialized. Maybe this case is not the best, I tried to find one, but it's controlled outside.